Hunting Queries/SecurityEvent/ProcessEntropy.txt

```diff
@@ -0,0 +1,75 @@
+// Name: Entropy for Processes for a given Host
+// Description: Entropy calculation used to help identify Hosts where they have a high variety of processes(a high entropy process list on a given Host over time).
+// This helps us identify rare processes on a given Host. Rare here means a process shows up on the Host relatively few times in the the last 7days.
+// The Weight is calculated based on the Entropy, Process Count and Distinct Hosts with that Process. The lower the Weight/ProcessEntropy the, more interesting.
+// The Weight calculation increases the Weight if the process executes more than once on the Host or has executed on more than 1 Hosts.
+// In general, this should identify processes on a Host that are rare and rare for the environment.
+//
+// References: https://medium.com/udacity/shannon-entropy-information-gain-and-picking-balls-from-buckets-5810d35d54b4
+// https://en.wiktionary.org/wiki/Shannon_entropy
+//
+// Id: 05208917-82de-46f7-a190-a65739a690f4
+//
+// DataSource: ##SecurityEvent
+//
+// Tactics: #Execution
+//
+// May need to reduce the number of days if the environment is very large.  Try 3-5 days, but less accuracy.
+let end = startofday(now());
+let start = end - 7d;
+let Exclude = SecurityEvent
+// Timeframe is set so that results do not change during the same day (UTC time)
+| where TimeGenerated >= start and TimeGenerated <= end
+| where EventID == 4688
+| summarize count() by Process 
+// Removing noisy processes for an environment, adjust as needed
+| where count_ >= 100000;
+let AllSecEvents = SecurityEvent
+| where TimeGenerated >= start and TimeGenerated <= end
+| where EventID == 4688
+| where Process !in~ ("conhost.exe")
+| project Computer, Process;
+// Removing noisy process from full list
+let Include = AllSecEvents | join kind= anti (
+Exclude
+) on Process;
+// Identifying prevalence for a given process in the environment
+let DCwPC = Include
+| join kind=inner (
+Include | summarize DistinctComputersWithProcessCount = dcount(Computer) by Process
```

Comment:
ditto - the summarized version of the table will always be smaller -- so put this on left hand side of the join (Include | summarize... by Process) | join kind=inner Include

Reply:
Done